sqlserver/datadog_checks/sqlserver/schemas.py

```diff
@@ -0,0 +1,417 @@
+    PARTITIONS_QUERY,
+    SCHEMA_QUERY,
+    TABLES_IN_SCHEMA_QUERY,
+) 
+
+
+from datadog_checks.sqlserver.utils import execute_query_output_result_as_dicts, get_list_chunks, is_azure_sql_database
+
+
+class SubmitData:
+
+    def __init__(self, submit_data_function, base_event, logger):
+        self._submit_to_agent_queue = submit_data_function
+        self._base_event = base_event
+        self._log = logger
+
+        self._columns_count = 0
+        self.db_to_schemas = {}  # dbname : { id : schema }
+        self.db_info = {}  # name to info
+
+    def set_base_event_data(self, hostname, tags, cloud_metadata, dbms_version):
+        self._base_event["host"] = hostname
+        self._base_event["tags"] = tags
+        self._base_event["cloud_metadata"] = cloud_metadata
+        self._base_event["dbms_version"] = dbms_version
+
+    def reset(self):
+        self._columns_count = 0
+        self.db_to_schemas.clear()
+        self.db_info.clear()
+
+    def store_db_infos(self, db_infos):
+        for db_info in db_infos:
+            self.db_info[db_info['name']] = db_info
+
+    def store(self, db_name, schema, tables, columns_count):
+        self._columns_count += columns_count
+        schemas = self.db_to_schemas.setdefault(db_name, {})
+        if schema["id"] in schemas:
+            known_tables = schemas[schema["id"]].setdefault("tables", [])
+            known_tables = known_tables.extend(tables)
+        else:
+            schemas[schema["id"]] = schema
+            schemas[schema["id"]]["tables"] = tables
+
+    def columns_since_last_submit(self):
+        return self._columns_count
+
+    def truncate(self, json_event):
+        max_length = 1000
+        if len(json_event) > max_length:
+            return json_event[:max_length] + " ... (truncated)"
+        else:
+            return json_event
+
+    # NOTE: DB with no schemas is never submitted
+    def submit(self):
+        if not self.db_to_schemas:
+            return
+        self._columns_count = 0
+        event = {**self._base_event, "metadata": [], "timestamp": time.time() * 1000}
+        for db, schemas_by_id in self.db_to_schemas.items():
+            db_info = {}
+            if db not in self.db_info:
+                self._log.error("Couldn't find database info for %s", db)
+                db_info["name"] = db
+            else:
+                db_info = self.db_info[db]
+            event["metadata"] = event["metadata"] + [{**(db_info), "schemas": list(schemas_by_id.values())}]
+        json_event = json.dumps(event, default=default_json_event_encoding)
+        self._log.debug("Reporting the following payload for schema collection: {}".format(self.truncate(json_event)))
+        self._submit_to_agent_queue(json_event)
+        self.db_to_schemas.clear()
+
+
+def agent_check_getter(self):
+    return self._check
+
+
+class Schemas(DBMAsyncJob):
+
+    # Requests for infromation about tables are done for a certain amount of tables at the time
```

Comment:
`infromation` -> `information`

Reply:
I actually removed this comment as a whole